app/src/main/java/io/github/mohamed/sallam/awb/LockService.java

```diff
@@ -46,13 +46,13 @@
  * @author Mohamed Sallam
  */
 public class LockService extends Service {
-    private final DetoxPeriodRepository detoxPeriodRepository;
+    private final DetoxPeriodRepositoryImpl detoxPeriodRepository;
```

Comment:
```suggestion
    private final DetoxPeriodRepository detoxPeriodRepository;
```
I think this the use of the interface.
Don't forget to import its lib.